stdlib/Profile/src/Profile.jl

```diff
@@ -808,7 +808,7 @@ end
 # Similar to `define_default_editors` in `Base.Filesystem` but for creating URIs not commands
 function editor_link(path::String, linenum::Int)
     editor = get(ENV, "JULIA_EDITOR", "")
-
+    path = escape_string(path)
```

Comment:
Is this really sufficient? The link created here is a form of URL, so should be URL encoded, no?

Reply:
I'm not sure we can do better in Profile without rolling our own [`HTTP.escapeuri` ](https://github.com/JuliaWeb/HTTP.jl/blob/382b6c3f8beac54151a5eba2a482aa5a0ccf4fab/src/URIs.jl#L323-L339) which might be ok?

Reply:
That `escapeuri` looks pretty equivalent to `encode_uri_component` in #55454, perhaps it could be worth extracting that out as a top-level function if we want to re-use it in a few other places too?

Reply:
Sounds good.